src/PatientChecking/PatientChecking/Feature/Patient/Queries/GetPatientInDetailByIdQuery.cs

```diff
@@ -0,0 +1,80 @@
+﻿using MediatR;
+using PatientChecking.ServiceModels;
+using PatientChecking.ServiceModels.Enum;
+using PatientChecking.Services.AppConfiguration;
+using PatientChecking.Services.Patient;
+using PatientChecking.Views.ViewModels;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace PatientChecking.Feature.Patient.Queries
+{
+    public class GetPatientInDetailByIdQuery : IRequest<PatientDetailViewModel>
+    {
+        public int PatientId { get; set; }
+        class GetPatientInDetailByIdQueryHandler : IRequestHandler<GetPatientInDetailByIdQuery, PatientDetailViewModel>
+        {
+            private readonly IPatientService _patientService;
+            private readonly IAppConfigurationService _provinceCityService;
+            public GetPatientInDetailByIdQueryHandler(IPatientService patientService, IAppConfigurationService provinceCityService)
+            {
+                _patientService = patientService;
+                _provinceCityService = provinceCityService;
+            }
+            public async Task<PatientDetailViewModel> Handle(GetPatientInDetailByIdQuery request, CancellationToken cancellationToken)
+            {
+                var cities = await _provinceCityService.GetProvinceCities();
+                var cityList = new List<string>();
+
+                foreach (PatientCheckIn.DataAccess.Models.ProvinceCity p in cities)
+                {
+                    cityList.Add(p.ProvinceCityName);
+                }
+
+                if (request.PatientId < 0)
+                {
+                    var emptyModel = new PatientDetailViewModel
+                    {
+                        PatientId = -1,
+                        PatientIdentifier = "",
+                        Nationality = "Vietnamese",
+                        ProvinceCities = cityList
+                    };
+                    return emptyModel;
+                }
+
+                var result = await _patientService.GetPatientInDetail(request.PatientId);
+
+                var model = new PatientDetailViewModel
+                {
+                    PatientId = result.PatientId,
```

Comment:
Just a suggestion: In the future, we can use `AutoMapper` for mapping these model. The mapping profile can be reused for other places.

Reply:
Thanks for your suggestion. I will apply it in the future